agent/src/softwarecontaineragent.h

```diff
@@ -79,7 +80,7 @@ class SoftwareContainerAgentError : public SoftwareContainerError
 class SoftwareContainerAgent
 {
     LOG_DECLARE_CLASS_CONTEXT("SCA", "SoftwareContainerAgent");
-    typedef std::shared_ptr<SoftwareContainer> SoftwareContainerPtr;
+    typedef std::shared_ptr<ContainerAgentInterface> SoftwareContainerPtr;
```

Comment:
I see. In that case I think `SoftwareContainerAbstractInterface` is better, yes.